File: src/main/java/gg/mineral/practice/kit/Kit.java
package gg.mineral.practice.kit;

import org.bukkit.inventory.ItemStack;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
public class Kit {
	private final String name;
	@Setter
	private ItemStack[] contents = new ItemStack[36], armourContents = new ItemStack[4];

	public Kit(ItemStack[] contents, ItemStack[] armourContents) {
		this("Custom", contents, armourContents);
	}

	public Kit(Kit kit) {
		this.name = kit.getName();
		this.contents = kit.getContents();
		this.armourContents = kit.getArmourContents();
	}
}

File: src/main/kotlin/gg/mineral/practice/kit/Kit.kt
package gg.mineral.practice.kit

import org.bukkit.inventory.ItemStack

class Kit(
    val name: String = "Custom",
    var contents: Array<ItemStack?> = arrayOfNulls(36),
    var armourContents: Array<ItemStack?> = arrayOfNulls(4)
) {
    constructor(kit: Kit) : this(kit.name, kit.contents, kit.armourContents)

    constructor(
        contents: Array<ItemStack?> = arrayOfNulls(36),
        armourContents: Array<ItemStack?> = arrayOfNulls(4)
    ) : this("Custom", contents, armourContents)
}
